cli/src/config.rs

```diff
@@ -34,42 +34,55 @@ fn extend_cli_args(
 
     let mut config_opts = Vec::new();
 
-    for (key, value) in read_config_file(path)?.into_iter() {
-        let opt = format!("--{}", key);
+    let mut port_opts = ::std::collections::HashMap::new();
+    port_opts.insert("port", 20222);
+    port_opts.insert("rpc-port", 8086);
+    port_opts.insert("ws-port", 8087);
```

Comment:
```suggestion
    let mut port_opts = HashMap::new();
    port_opts.insert("port", 20222);
    port_opts.insert("rpc-port", 8086);
    port_opts.insert("ws-port", 8087);
```

Reply:
I recommend passing the default opts from the argument of `extend_cli_args`.

maybe like:

```
fn extend_cli_args(
    cli_args: Vec<String>,
    path: Option<&Path>,
    default_opts: HashMap<String, String>,
)  -> Result<Vec<String>, Box<dyn std::error::Error>> {...}

fn default_opts() -> HashMap<String, String> {
    let mut opts = HashMap::new();
    map.insert("port".into(), 20222.to_string());
    map.insert("rpc-port".into(), 8086.to_string());
    map.insert("ws-port".into(), 8087.to_string());
    map
}
```